### PR TITLE
fix(streaming): create mview progress report under concurrent checkpoint

### LINKED ISSUE #3598: bug: Barrier carry wrong create_mv msg from cn to meta(e2e_parallel_test_fail)
**Describe the bug**
We have been having a lot of failed in e2e_parallel_test_fail. We cannot query any things in mv.

WeChate97a86a773cc33db24df1a1953312231
And we can query the right result in the mv after some times. And @zwang28  find this bug easy to reproduce after concurrent checkpoint. https://github.com/singularity-data/risingwave/pull/2184

**To Reproduce**
@hzxa21 find the epoch of query large than the epoch of create_mv_over in frontend , But it will less than the epoch of mv written to hummock. So we think the cn tell frontend(or meta) a wrong epoch.
(write_to_hummock_epoch > query epoch > create_mv_over_epoch).

And we see the code of cn. we find how bugs are created

1. create_mv_barrier(epoch1) flows through all graph. but the mv is not finished.
2. when epoch2 flows through actor1, the mv is not finished, and barrier(epoch2)_collected is not finished too.
3. when epoch3 flows through actor1, the snapshot has all sent, so this actor is finished(ManagedBarrierState.progress.done = true).and barrier(epoch3)_collected is not finished too.
4. when epoch2 flows through actor2. barrier(epoch2)_collected is finish. So the barrier with `progress.done = true` send to meta, Then meta think this mv is over with epoch2. But it will be commit in epoch3. 
5. So query between epoch2 and epoch3 will wrong

**Additional context**

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As title. See #3598 for more details.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- #3598
- #1156

### CODE REVIEW

File: src/stream/src/task/barrier_manager/managed_state.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ enum ManagedBarrierStateInner {
 pub(super) struct ManagedBarrierState {
     epoch_barrier_state_map: HashMap<u64, ManagedBarrierStateInner>,
 
-    pub create_mview_progress: HashMap<ActorId, ChainState>,
+    pub(super) create_mview_progress: HashMap<u64, HashMap<ActorId, ChainState>>,
```
Comment (MrCroxx) [inline src/stream/src/task/barrier_manager/managed_state.rs:50]: Would you please add a comment to explain the mapping? ❤️

**Review by MrCroxx — APPROVED** [review]
Rest LGTM ❤️

**Review by xxhZs — COMMENTED** [review]
LGTM, I will test it with https://github.com/singularity-data/risingwave/pull/3511

**Review by hzxa21 — APPROVED** [review]
LGTM. Thanks for the quick fix!

**Review by yezizp2012 — APPROVED** [review]

**Comment by BugenZhao** [comment]
> LGTM, I will test it with #3511

Cool. You may merge this PR after tested.

**Review by skyzh — APPROVED** [review]

**Comment by BugenZhao** [comment]
Seems this PR doesn't fully resolve the issue. Will remove "close" in PR body.

**Pull request merged**